BuildTasks/Common/Common.ts

```diff
@@ -82,11 +82,19 @@ export function setTfxManifestArguments(tfx: ToolRunner): (() => void) {
         if (isPreview) { jsonOverrides.galleryFlags = ["Preview"]; };
     }
 
-    const extensionVersion = tl.getInput("extensionVersion", false);
+    let extensionVersion = tl.getInput("extensionVersion", false);
     if (extensionVersion) {
-        tl.debug(`Overriding extension version to: ${extensionVersion}`);
-        jsonOverrides = (jsonOverrides || {});
-        jsonOverrides.version = extensionVersion;
+       const extractedVersions = extensionVersion.match(/[0-9]+\.[0-9]+\.[0-9]+/);
+       if (extractedVersions && extractedVersions.length === 1) {
+            extensionVersion = extractedVersions[0];
+            tl.debug(`Overriding extension version to: ${extensionVersion}`);
+            jsonOverrides = (jsonOverrides || {});
+            jsonOverrides.version = extensionVersion;
+        }
+        else
```

Comment:
As for the 4 part version number. It's against guidelines, against semver, against the version that can be stored in the build tasks... Should it not be better to have people stay within the standard, instead of adding a feature that will cause unwanted behavior?

What 3 parts of the version should we pass to the build tasks in this case? Or should we raise an error/warning for tasks containing build tasks?